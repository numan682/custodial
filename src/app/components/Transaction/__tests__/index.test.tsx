import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { selectTicker } from 'app/state/network/selectors'
import { when } from 'jest-when'
import * as React from 'react'
import { Provider, useSelector } from 'react-redux'
import { configureAppStore } from 'store/configureStore'
import { backend } from 'vendors/backend'
import { BackendAPIs } from 'config'
import copy from 'copy-to-clipboard'

import { Transaction } from '..'
import * as transactionTypes from 'app/state/transaction/types'
import { NetworkType } from 'app/state/network/types'
import type { UseTranslationResponse, Trans } from 'react-i18next'

jest.mock('copy-to-clipboard')

type TransType = typeof Trans
jest.mock('react-i18next', () => ({
  Trans: (({ i18nKey }) => <>{i18nKey}</>) as TransType,
  useTranslation: () => {
    return {
      t: (str: string) => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    } as UseTranslationResponse<'translation', 'translation'>
  },
}))

jest.mock('vendors/backend')
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: jest.fn(),
}))

const renderComponent = (
  store: any,
  ref: any,
  transaction: transactionTypes.Transaction,
  network: NetworkType,
) =>
  render(
    <Provider store={store}>
      <Transaction referenceAddress={ref} transaction={transaction} network={network} />
    </Provider>,
  )

describe('<Transaction  />', () => {
  let store: ReturnType<typeof configureAppStore>
  const ref = 'sourceAddr'
  const transaction = {
    amount: 1000000n.toString(),
    timestamp: 1618018255000,
    from: 'source',
    to: 'destination',
    type: transactionTypes.TransactionType.StakingTransfer,
    hash: 'ff1234',
    status: true,
  } as transactionTypes.Transaction
  const network = 'mainnet'

  beforeEach(() => {
    jest.mocked(backend).mockImplementation(() => BackendAPIs.OasisScan)
    store = configureAppStore()

    when(useSelector as any)
      .calledWith(selectTicker)
      .mockReturnValue('TEST')
  })

  it('should match snapshot', () => {
    const component = renderComponent(store, ref, transaction, network)
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should copy an address and tx hash value to clipboard', async () => {
    renderComponent(store, ref, transaction, network)

    await userEvent.click(screen.getByLabelText('ContactInfo'))
    expect(copy).toHaveBeenNthCalledWith(1, 'source')

    await userEvent.click(screen.getByLabelText('Package'))
    expect(copy).toHaveBeenNthCalledWith(2, 'ff1234')
  })

  it('should mark failed transactions', () => {
    renderComponent(store, ref, { ...transaction, status: false }, network)
    expect(screen.getByText('account.transaction.failed')).toBeInTheDocument()
  })

  it('should handle unknown transaction types gracefully', () => {
    const component = renderComponent(
      store,
      'sourceAddr',
      {
        amount: 1000000n.toString(),
        timestamp: 1618018255000,
        from: 'source',
        to: 'destination',
        type: 'turboencabulate' as transactionTypes.TransactionType,
        hash: 'ff1234',
        fee: undefined,
        level: undefined,
        status: true,
        runtimeName: undefined,
        runtimeId: undefined,
        round: undefined,
      },
      network,
    )
    expect(component.container.firstChild).toMatchSnapshot()
  })

  it('should not render a link when address is undefined', () => {
    renderComponent(
      store,
      'sourceAddr',
      {
        amount: 1000000n.toString(),
        timestamp: 1618018255000,
        from: 'sourceAddr',
        to: undefined,
        type: 'anyType' as transactionTypes.TransactionType,
        hash: 'ff1234',
        fee: undefined,
        level: undefined,
        status: true,
        runtimeName: undefined,
        runtimeId: undefined,
        round: undefined,
      },
      network,
    )

    expect(screen.queryByTestId('external-wallet-address')).not.toBeInTheDocument()
    expect(screen.getByText('common.unavailable')).toBeInTheDocument()
  })

  it('should render testnet link', () => {
    renderComponent(store, ref, transaction, 'testnet')
    expect(screen.getByTestId('explorer-link')).toHaveAttribute(
      'href',
      'https://testnet.oasisscan.com/transactions/ff1234',
    )
  })

  it('should render monitor link', () => {
    jest.mocked(backend).mockImplementation(() => BackendAPIs.OasisMonitor)
    renderComponent(store, ref, transaction, network)
    expect(screen.getByTestId('explorer-link')).toHaveAttribute(
      'href',
      'https://oasismonitor.com/operation/ff1234',
    )
  })

  it('should render testnet monitor link', () => {
    jest.mocked(backend).mockImplementation(() => BackendAPIs.OasisMonitor)
    renderComponent(store, ref, transaction, 'testnet')
    expect(screen.getByTestId('explorer-link')).toHaveAttribute(
      'href',
      'https://testnet.oasismonitor.com/operation/ff1234',
    )
  })

  it('should render paratime transaction row', () => {
    const runtimeId = '123'

    renderComponent(
      store,
      'source',
      {
        ...transaction,
        runtimeId,
        runtimeName: 'Name',
        round: 951933,
        type: transactionTypes.TransactionType.ConsensusDeposit,
      },
      network,
    )

    expect(screen.getByText('common.round')).toBeInTheDocument()
    expect(screen.queryByText('common.block')).not.toBeInTheDocument()
    expect(screen.getByTestId('explorer-link')).toHaveAttribute(
      'href',
      `https://oasisscan.com/paratimes/transactions/ff1234?runtime=${runtimeId}`,
    )
    expect(screen.getByLabelText('Inherit')).toHaveStyleRule('stroke', '#FFCA58')
  })
})
