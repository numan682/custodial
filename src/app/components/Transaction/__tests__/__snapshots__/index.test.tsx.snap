// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Transaction  /> should handle unknown transaction types gracefully 1`] = `
.c2 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c2 g {
  fill: inherit;
  stroke: inherit;
}

.c2 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c2 *[stroke*='#'],.c2 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c2 *[fill-rule],
.c2 *[FILL-RULE],
.c2 *[fill*='#'],.c2 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c12 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 20px;
  height: 20px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c12 g {
  fill: inherit;
  stroke: inherit;
}

.c12 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c12 *[stroke*='#'],.c12 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c12 *[fill-rule],
.c12 *[FILL-RULE],
.c12 *[fill*='#'],.c12 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
  border-radius: 6px;
  overflow: hidden;
  box-shadow: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: solid 1px background-front-border;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-bottom: 12px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 75%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px;
  cursor: pointer;
}

.c10:hover {
  background-color: rgba(51,51,51,0.06274509803921569);
  color: #000000;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 6px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 6px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 25%;
  padding-right: 12px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c21 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c21:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fit,minmax(min(96px,100%),1fr) );
  grid-gap: 0px 0px;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c14 {
  font-size: 18px;
  line-height: 24px;
}

.c17 {
  font-size: 26px;
  line-height: 32px;
  font-weight: bold;
}

.c18 {
  font-size: 12px;
  line-height: 18px;
  color: lightText;
  font-weight: 600;
}

.c19 {
  font-size: 14px;
  line-height: 20px;
  color: successful-label;
  font-weight: bold;
}

.c22 {
  font-size: 14px;
  line-height: 20px;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    border-radius: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    border-bottom: solid 1px background-front-border;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding-right: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    margin-top: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    width: 6px;
  }
}

@media only screen and (min-width:768px) and (max-width:1024px) {
  .c6 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }

  .c6 > * {
    width: 100%;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <svg
      aria-label="Alert"
      class="c2"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 17v2m0-9v6m0-13L2 22h20L12 3z"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
    <div
      class="c3"
    />
    <span
      class="c4"
    >
      account.transaction.unrecognizedTransaction.header
    </span>
  </header>
  <div
    class="c5 c6"
  >
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <div
            class="c10"
            tabindex="0"
          >
            <div
              class="c11"
            >
              <svg
                aria-label="ContactInfo"
                class="c12"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 12a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm4 6v-2c0-2.25-1.787-4-4.036-4h.054C2.768 12 1 13.75 1 16v2M12 7h12M12 17h10m-10-5h7"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="c3"
            />
            <div
              class="c13"
            >
              <span
                class="c4"
              >
                account.transaction.unrecognizedTransaction.destination
              </span>
              <span
                class="c14"
              >
                source
              </span>
            </div>
          </div>
          <div
            class="c10"
            tabindex="0"
          >
            <div
              class="c11"
            >
              <svg
                aria-label="Package"
                class="c12"
                viewBox="0 0 24 24"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.371.571 12 .423l-.371.148-10 4L1 4.823v14.354l.629.251 10 4 .371.149.371-.149 10-4 .629-.251V4.823l-.629-.252-10-4zM3 6.977v10.846l8 3.2V10.177l-8-3.2zm10 3.2v10.846l8-3.2V6.977l-8 3.2zM19.307 5.5 12 2.577 9.943 3.4l7.307 2.923 2.057-.823zm-14.614 0L7.25 4.477 14.557 7.4 12 8.423 4.693 5.5z"
                  fill="#000"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="c3"
            />
            <div
              class="c13"
            >
              <span
                class="c4"
              >
                common.hash
              </span>
              <span
                class="c14"
              >
                ff1234
              </span>
            </div>
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c15"
          >
            <div
              class="c11"
            >
              <svg
                aria-label="Clock"
                class="c12"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 7a1 1 0 1 0-2 0v5a1 1 0 0 0 .4.8l4 3a1 1 0 0 0 1.2-1.6L13 11.5V7z"
                  fill="#000"
                />
                <path
                  clip-rule="evenodd"
                  d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM3 12a9 9 0 1 1 18 0 9 9 0 0 1-18 0z"
                  fill="#000"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="c3"
            />
            <div
              class="c13"
            >
              <span
                class="c4"
              >
                common.time
              </span>
              <span
                class="c14"
              >
                4/10/2021, 1:30:55 AM
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c16"
    >
      <span
        class="c17"
      >
        <span>
          <span>
            0.001
          </span>
          <span
            class="c18"
          >
            <span
              class="notranslate"
              translate="no"
            >
               TEST
            </span>
          </span>
        </span>
      </span>
      <span
        class="c19"
      >
        <span>
          account.transaction.successful
        </span>
      </span>
    </div>
  </div>
  <footer
    class="c20"
  >
    <a
      class="c21"
      data-testid="explorer-link"
      href="https://oasisscan.com/transactions/ff1234"
      rel="noopener"
      target="_blank"
    >
      <span
        class="c22"
      >
        account.transaction.explorerLink
      </span>
      <svg
        aria-label="FormNext"
        class="c12"
        viewBox="0 0 24 24"
      >
        <path
          d="m9 6 6 6-6 6"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </a>
  </footer>
</div>
`;

exports[`<Transaction  /> should match snapshot 1`] = `
.c2 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 24px;
  height: 24px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c2 g {
  fill: inherit;
  stroke: inherit;
}

.c2 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c2 *[stroke*='#'],.c2 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c2 *[fill-rule],
.c2 *[FILL-RULE],
.c2 *[fill*='#'],.c2 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c12 {
  display: inline-block;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 20px;
  height: 20px;
  fill: #7D4CDB;
  stroke: #7D4CDB;
}

.c12 g {
  fill: inherit;
  stroke: inherit;
}

.c12 *:not([stroke])[fill='none'] {
  stroke-width: 0;
}

.c12 *[stroke*='#'],.c12 *[STROKE*='#'] {
  stroke: inherit;
  fill: none;
}

.c12 *[fill-rule],
.c12 *[FILL-RULE],
.c12 *[fill*='#'],.c12 *[FILL*='#'] {
  fill: inherit;
  stroke: none;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  background-color: #FFFFFF;
  color: #444444;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 12px;
  border-radius: 6px;
  overflow: hidden;
  box-shadow: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: solid 1px background-front-border;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-bottom: 12px;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-bottom: 12px;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 75%;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 0px;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px;
  cursor: pointer;
}

.c10:hover {
  background-color: rgba(51,51,51,0.06274509803921569);
  color: #000000;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding-right: 6px;
}

.c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 12px;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  margin-top: 6px;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 25%;
  padding-right: 12px;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 0;
  min-height: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  width: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c3 {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
  width: 12px;
}

.c21 {
  box-sizing: border-box;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
}

.c21:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c8 {
  display: grid;
  box-sizing: border-box;
  grid-template-columns: repeat( auto-fit,minmax(min(96px,100%),1fr) );
  grid-gap: 0px 0px;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  font-weight: bold;
}

.c14 {
  font-size: 18px;
  line-height: 24px;
}

.c17 {
  font-size: 26px;
  line-height: 32px;
  font-weight: bold;
}

.c18 {
  font-size: 12px;
  line-height: 18px;
  color: lightText;
  font-weight: 600;
}

.c19 {
  font-size: 14px;
  line-height: 20px;
  color: successful-label;
  font-weight: bold;
}

.c22 {
  font-size: 14px;
  line-height: 20px;
}

.c6 {
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

@media only screen and (max-width:768px) {
  .c0 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c0 {
    border-radius: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    border-bottom: solid 1px background-front-border;
  }
}

@media only screen and (max-width:768px) {
  .c1 {
    padding-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c5 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c9 {
    padding: 0px;
  }
}

@media only screen and (max-width:768px) {
  .c10 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c11 {
    padding-right: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c15 {
    padding: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    margin-top: 3px;
  }
}

@media only screen and (max-width:768px) {
  .c16 {
    padding-right: 6px;
  }
}

@media only screen and (max-width:768px) {
  .c3 {
    width: 6px;
  }
}

@media only screen and (min-width:768px) and (max-width:1024px) {
  .c6 {
    -webkit-flex-direction: column-reverse;
    -ms-flex-direction: column-reverse;
    flex-direction: column-reverse;
    -webkit-align-items: flex-end;
    -webkit-box-align: flex-end;
    -ms-flex-align: flex-end;
    align-items: flex-end;
  }

  .c6 > * {
    width: 100%;
  }
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <svg
      aria-label="LinkPrevious"
      class="c2"
      viewBox="0 0 24 24"
    >
      <path
        d="M22 12H2m9-9-9 9 9 9"
        fill="none"
        stroke="#000"
        stroke-width="2"
      />
    </svg>
    <div
      class="c3"
    />
    <span
      class="c4"
    >
      account.transaction.transfer.received
    </span>
  </header>
  <div
    class="c5 c6"
  >
    <div
      class="c7"
    >
      <div
        class="c8"
      >
        <div
          class="c9"
        >
          <div
            class="c10"
            tabindex="0"
          >
            <div
              class="c11"
            >
              <svg
                aria-label="ContactInfo"
                class="c12"
                viewBox="0 0 24 24"
              >
                <path
                  d="M5 12a3 3 0 1 1 0-6 3 3 0 0 1 0 6zm4 6v-2c0-2.25-1.787-4-4.036-4h.054C2.768 12 1 13.75 1 16v2M12 7h12M12 17h10m-10-5h7"
                  fill="none"
                  stroke="#000"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="c3"
            />
            <div
              class="c13"
            >
              <span
                class="c4"
              >
                common.from
              </span>
              <span
                class="c14"
              >
                source
              </span>
            </div>
          </div>
          <div
            class="c10"
            tabindex="0"
          >
            <div
              class="c11"
            >
              <svg
                aria-label="Package"
                class="c12"
                viewBox="0 0 24 24"
              >
                <path
                  clip-rule="evenodd"
                  d="M12.371.571 12 .423l-.371.148-10 4L1 4.823v14.354l.629.251 10 4 .371.149.371-.149 10-4 .629-.251V4.823l-.629-.252-10-4zM3 6.977v10.846l8 3.2V10.177l-8-3.2zm10 3.2v10.846l8-3.2V6.977l-8 3.2zM19.307 5.5 12 2.577 9.943 3.4l7.307 2.923 2.057-.823zm-14.614 0L7.25 4.477 14.557 7.4 12 8.423 4.693 5.5z"
                  fill="#000"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="c3"
            />
            <div
              class="c13"
            >
              <span
                class="c4"
              >
                common.hash
              </span>
              <span
                class="c14"
              >
                ff1234
              </span>
            </div>
          </div>
        </div>
        <div
          class="c9"
        >
          <div
            class="c15"
          >
            <div
              class="c11"
            >
              <svg
                aria-label="Clock"
                class="c12"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13 7a1 1 0 1 0-2 0v5a1 1 0 0 0 .4.8l4 3a1 1 0 0 0 1.2-1.6L13 11.5V7z"
                  fill="#000"
                />
                <path
                  clip-rule="evenodd"
                  d="M12 1C5.925 1 1 5.925 1 12s4.925 11 11 11 11-4.925 11-11S18.075 1 12 1zM3 12a9 9 0 1 1 18 0 9 9 0 0 1-18 0z"
                  fill="#000"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
            <div
              class="c3"
            />
            <div
              class="c13"
            >
              <span
                class="c4"
              >
                common.time
              </span>
              <span
                class="c14"
              >
                4/10/2021, 1:30:55 AM
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c16"
    >
      <span
        class="c17"
      >
        <span>
          <span>
            0.001
          </span>
          <span
            class="c18"
          >
            <span
              class="notranslate"
              translate="no"
            >
               TEST
            </span>
          </span>
        </span>
      </span>
      <span
        class="c19"
      >
        <span>
          account.transaction.successful
        </span>
      </span>
    </div>
  </div>
  <footer
    class="c20"
  >
    <a
      class="c21"
      data-testid="explorer-link"
      href="https://oasisscan.com/transactions/ff1234"
      rel="noopener"
      target="_blank"
    >
      <span
        class="c22"
      >
        account.transaction.explorerLink
      </span>
      <svg
        aria-label="FormNext"
        class="c12"
        viewBox="0 0 24 24"
      >
        <path
          d="m9 6 6 6-6 6"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </a>
  </footer>
</div>
`;
